### DISCUSSION TITLE
Binding not working

### DISCUSSION THREAD

**Discussion by mistral61** [ng-matero/ng-matero#618] [Q&A]
I started from the demo site, deleting the unuseful things.
I have a page similar to the remote-data sample, i  changed my backend in order to have an identical answer, in the subscribe i can see that the data are correctly coming and i set the mtx-grid [data] = to the array from my backend (a .NET core 6 webAPI connected to MariaDB, but it should not be the problem), 2 records.
Practically my code is identical to the one of the sample site, apart the fact that does not connect to github but to a local .NET webapi  done with Visual Studio 2022: but i repeat, in subscribe the data array is right.
But the mtx-grid hangs: it takes many seconds in order to see the data; curiously if i resize the browser window the data are immediately displayed .
If i provide mocked data these are instantly displayed.
What can be the cause of this strange behavior ? Matero is ^16.2.0

**Comment by nzbin** [comment]
Please provide an online example

**Reply by mistral61** [reply]
Hello, the problem is resolved with a ChangeDetectionRef.
The changes:
![image](https://github.com/ng-matero/ng-matero/assets/5938930/1e5e2e79-5421-4a49-9005-b5bbf01022f2)
with the markForCheck it works perfectly, no more hanging.
The only thing that i don't understand how the remote Matero demo works without using this trick.
I see that ChangeDetectorRef is used in mobile-sidenav and user components, that seems unrelated to the remote date component.
